mlflow/sagemaker/container/__init__.py

```diff
@@ -97,6 +104,96 @@ def _serve():
     _sigterm_handler(nginx.pid, gunicorn.pid)
 
 
+def _get_conda_env(model_config):
+    model_py_version = None
+    if pyfunc.PY_VERSION in model_config:
+        model_py_version = model_config[pyfunc.PY_VERSION]
+
+    if pyfunc.ENV in model_config:
+        print_flush("Activating custom Anaconda environment")
+        env = model_config[pyfunc.ENV]
+        env_path_src = os.path.join("/opt/ml/model", env)
+        _add_py_version_to_conda_env_if_necessary(env_path_src, model_py_version)
+        env_path_dst = os.path.join("/opt/mlflow/", env)
+        env_path_dst_dir = os.path.dirname(env_path_dst)
+        if not os.path.exists(env_path_dst_dir):
+            os.makedirs(env_path_dst_dir)
+        # TODO: should we test that the environment does not include any of the server dependencies?
+        # Those are gonna be reinstalled. should probably test this on the client side
+        shutil.copyfile(env_path_src, env_path_dst)
+        return CondaEnv(name="custom_env", config=env, py_version=None)
+    elif model_py_version is None:
+        print_flush("The model does not specify a Python version or a custom Anaconda environment"
+                    " to use.")
+        _warn_potentially_incompatible_conda_env()
+        return None
+    elif not _conda_supports_py_version(model_py_version):
+        print_flush("The version of python used to serialize the model, Python {mpyv}, is not"
+                     " supported by Anaconda.".format(mpyv=model_py_version))
+        _warn_potentially_incompatible_conda_env()
+        return None
+    elif model_py_version == PYTHON_VERSION:
+        print_flush("The model's Python version matches the default environment's"
+                    " Python version: {dpyv}. Using the default environment.".format(
+                        dpyv=DEFAULT_CONDA_PYTHON_VERSION))
+        return None
+    else:
+        print_flush("The model's Python version is {mpyv}. Activating Anaconda environment with"
+                    " Python version {mpyv}.".format(mpyv=model_py_version))
+        return CondaEnv(name="py_env", config=None, py_version=model_py_version)
+
+
+def _add_py_version_to_conda_env_if_necessary(env_path, model_py_version):
+    try:
+        with open(env_path, "r") as f:
+            env_yaml = yaml.load(f)
+
+        deps = env_yaml["dependencies"]
+        env_py_versions = list(filter(
+            lambda item : "=" in item and item.split("=")[0].lower() == "python", deps))
+        env_contains_python = len(env_py_versions) > 0
+        if env_contains_python:
+            env_py_version = env_py_versions[-1].split("=")[1]
+            print_flush("Using Python version specified by the custom Anaconda environment:" 
+                        " Python {epyv}.".format(epyv=env_py_version))
```

Comment:
According to the documentation here:
https://conda.io/docs/user-guide/tasks/build-packages/package-spec.html

It seems spaces may exist instead of/in addition to equal signs.

Reply:
Do we depend on `conda`? We may be able to leverage MatchSpec to help deal with these peculiarities:
https://github.com/conda/conda/blob/ce207022f257019cca55a79d025d9e324591bf0b/conda/models/match_spec.py#L72